### PR TITLE
Various fixes in the reference code

### PR DESCRIPTION
**Pull request by ptomato**
Fixes small compliance issues in the reference code, that I came across while doing other things. Also removes unnecessary code from the parser tests.

### CODE REVIEW

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -2632,9 +2638,15 @@ export function GetAvailableNamedTimeZoneIdentifier(identifier) {
...
   if (canonicalTimeZoneIdsCache === undefined) {
     const canonicalTimeZoneIds = IntlSupportedValuesOf?.('timeZone');
-    canonicalTimeZoneIdsCache = canonicalTimeZoneIds
-      ? new Map(canonicalTimeZoneIds.map((id) => [ASCIILowercase(id), id]))
-      : null;
+    if (canonicalTimeZoneIds) {
+      canonicalTimeZoneIdsCache = new Map();
+      for (let ix = 0; ix < canonicalTimeZoneIds.length; ix++) {
+        const id = canonicalTimeZoneIds[ix];
+        canonicalTimeZoneIdsCache.set(ASCIILowercase(id), id);
```
Comment (gibson042) [inline polyfill/lib/ecmascript.mjs:2645]:
```suggestion
        Call(MapPrototypeSet, canonicalTimeZoneIdsCache, [ASCIILowercase(id), id]);
```

(or just `new OriginalMap(Call(ArrayPrototypeMap, canonicalTimeZoneIds, [(id) => [ASCIILowercase(id), id]]))`)

File: polyfill/lib/ecmascript.mjs [hunk 2]
```diff
@@ -2632,9 +2638,15 @@ export function GetAvailableNamedTimeZoneIdentifier(identifier) {
   // implementations lacking this API, set the cache to `null` to avoid retries.
   if (canonicalTimeZoneIdsCache === undefined) {
     const canonicalTimeZoneIds = IntlSupportedValuesOf?.('timeZone');
-    canonicalTimeZoneIdsCache = canonicalTimeZoneIds
-      ? new Map(canonicalTimeZoneIds.map((id) => [ASCIILowercase(id), id]))
-      : null;
+    if (canonicalTimeZoneIds) {
+      canonicalTimeZoneIdsCache = new Map();
```
Comment (gibson042) [inline polyfill/lib/ecmascript.mjs:2642]:
```suggestion
      canonicalTimeZoneIdsCache = new OriginalMap();
```

**Review by gibson042 — APPROVED** [review]

**Review by justingrant — APPROVED** [review]
👍

**Comment by ptomato** [comment]
Thanks for the reviews!

**Pull request merged**